SU2_CFD/include/iteration/CFluidIteration.hpp

```diff
@@ -0,0 +1,154 @@
+   */
+  CFluidIteration(const CConfig* config);
+
+  /*!
+   * \brief Destructor of the class.
+   */
+  ~CFluidIteration(void) override;
+
+  /*!
+   * \brief Preprocessing to prepare for an iteration of the physics.
+   * \param[in] ??? - Description here.
+   */
+  void Preprocess(COutput* output, CIntegration**** integration, CGeometry**** geometry, CSolver***** solver,
+                  CNumerics****** numerics, CConfig** config, CSurfaceMovement** surface_movement,
+                  CVolumetricMovement*** grid_movement, CFreeFormDefBox*** FFDBox, unsigned short val_iZone,
+                  unsigned short val_iInst) override;
+
+  /*!
+   * \brief Perform a single iteration of the fluid system.
+   * \param[in] output - Pointer to the COutput class.
+   * \param[in] integration - Container vector with all the integration methods.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver - Container vector with all the solutions.
+   * \param[in] numerics - Description of the numerical method (the way in which the equations are solved).
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] surface_movement - Surface movement classes of the problem.
+   * \param[in] grid_movement - Volume grid movement classes of the problem.
+   * \param[in] FFDBox - FFD FFDBoxes of the problem.
+   */
+  void Iterate(COutput* output, CIntegration**** integration, CGeometry**** geometry, CSolver***** solver,
+               CNumerics****** numerics, CConfig** config, CSurfaceMovement** surface_movement,
+               CVolumetricMovement*** grid_movement, CFreeFormDefBox*** FFDBox, unsigned short val_iZone,
+               unsigned short val_iInst) override;
+
+  /*!
+   * \brief Iterate the fluid system for a number of Inner_Iter iterations.
+   * \param[in] output - Pointer to the COutput class.
+   * \param[in] integration - Container vector with all the integration methods.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver - Container vector with all the solutions.
+   * \param[in] numerics - Description of the numerical method (the way in which the equations are solved).
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] surface_movement - Surface movement classes of the problem.
+   * \param[in] grid_movement - Volume grid movement classes of the problem.
+   * \param[in] FFDBox - FFD FFDBoxes of the problem.
+   */
+  void Solve(COutput* output, CIntegration**** integration, CGeometry**** geometry, CSolver***** solver,
+             CNumerics****** numerics, CConfig** config, CSurfaceMovement** surface_movement,
+             CVolumetricMovement*** grid_movement, CFreeFormDefBox*** FFDBox, unsigned short val_iZone,
+             unsigned short val_iInst) override;
+
+  /*!
+   * \brief Updates the containers for the fluid system.
+   * \param[in] ??? - Description here.
+   */
```

Comment:
I wonder if we need documentation strings on the derived classes, especially on this class family this stuff is out of date all the time.